pyserini/search/lucene/tprob/_search_tprob.py

```diff
@@ -0,0 +1,231 @@
+    def get_ibm_score(arguments):
+        query_text_lst = arguments['query_text_lst']
+        test_doc = arguments['test_doc']
+        searcher = arguments['searcher']
+        field_name = arguments['field_name']
+        source_lookup = arguments['source_lookup']
+        target_lookup = arguments['target_lookup']
+        tran = arguments['tran']
+        collect_probs = arguments['collect_probs']
+        max_sim = arguments['max_sim']
+
+        if searcher.documentRaw(test_doc) is None:
+            print(f'{test_doc} is not found in searcher')
+        document_text = json.loads(searcher.documentRaw(test_doc))[field_name]
+        doc_token_lst = document_text.split(" ")
+        total_query_prob = 0
+        doc_size = len(doc_token_lst)
+        query_size = len(query_text_lst)
+        for querytoken in query_text_lst:
+            target_map = {}
+            total_tran_prob = 0
+            collect_prob = collect_probs[querytoken]
+            max_sim_score = 0
+            if querytoken in target_lookup.keys():
+                query_word_id = target_lookup[querytoken]
+                if query_word_id in tran.keys():
+                    target_map = tran[query_word_id]
+                    for doctoken in doc_token_lst:
+                        tran_prob = 0
+                        doc_word_id = 0
+                        if doctoken in source_lookup.keys():
+                            doc_word_id = source_lookup[doctoken]
+                            if doc_word_id in target_map.keys():
+                                tran_prob = max(target_map[doc_word_id], tran_prob)
+                                max_sim_score = max(tran_prob, max_sim_score)
+                                total_tran_prob += (tran_prob/doc_size)
+            if max_sim:
+                query_word_prob = math.log(
+                    (1 - LAMBDA_VALUE) * max_sim_score + LAMBDA_VALUE * collect_prob)
+            else:
+                query_word_prob = math.log(
+                    (1 - LAMBDA_VALUE) * total_tran_prob + LAMBDA_VALUE * collect_prob)
+
+            total_query_prob += query_word_prob
+        return total_query_prob / query_size
+
+    def search(self, query_text, query_field_text, hits, max_sim):
+        self.bm25search.set_bm25(0.82, 0.68)
+
+        bm25_results = self.bm25search.search(query_text, hits)
+        origin_scores = [bm25_result.score for bm25_result in bm25_results]
+        test_docs = [bm25_result.docid for bm25_result in bm25_results]
+        if (test_docs == []):
+            print(query_text)
+
+        query_text_lst = query_field_text.split(' ')
+        total_term_freq = self.index_reader.getSumTotalTermFreq(self.field_name)
+        collect_probs = {}
+        for querytoken in query_text_lst:
+            collect_probs[querytoken] = max(self.index_reader.totalTermFreq(
+                JTerm(self.field_name, querytoken)) / total_term_freq,
+                MIN_COLLECT_PROB)
+
+        arguments = [{
```

Comment:
this is a bit odd that you're passing in args as a json internally inside Python... why do this?

Reply:
I've changed it into a list of tuple, should looks better now